docs/endpoint.md

```diff
@@ -0,0 +1,109 @@
+# Client Library Endpoint
```

Comment:
Would it be helpful to also mention which versions support customizing endpoints? For example, starting with `libraries-bom:x.x.x`